src/query-builder/QueryBuilder.ts

```diff
@@ -1048,6 +1065,78 @@ export abstract class QueryBuilder<Entity> {
         )
     }
 
+    protected createCteExpression(): string {
+        if (!this.hasCommonTableExpressions()) {
+            return ""
+        }
+        const databaseRequireRecusiveHint =
+            this.connection.driver.cteCapabilities.requiresRecursiveHint
+
+        const cteStrings = this.expressionMap.commonTableExpressions.map(
+            (cte) => {
+                const cteBodyExpression =
+                    typeof cte.queryBuilder === "string"
+                        ? cte.queryBuilder
+                        : cte.queryBuilder.getQuery()
+                if (typeof cte.queryBuilder !== "string") {
+                    if (cte.queryBuilder.hasCommonTableExpressions()) {
+                        throw new TypeORMError(
+                            `Nested CTEs aren't supported (CTE: ${cte.alias})`,
+                        )
+                    }
+                    if (
+                        !this.connection.driver.cteCapabilities.writable &&
+                        !(cte.queryBuilder instanceof SelectQueryBuilder)
```

Comment:
replace it with `InstanceChecker.isSelectQueryBuilder`. From 0.3.0 we avoid using `instanceof` because it doesn't play well in the monorepo projects with multiple instances of TypeORM across different node_modules.

Reply:
Done.